src/Native/Runtime/amd64/ExceptionHandling.S

```diff
@@ -0,0 +1,525 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+.intel_syntax noprefix
+#include <unixasmmacros.inc>
+#include <AsmOffsets.inc>         // generated by the build from AsmOffsets.cpp
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//
+// RhpThrowHwEx
+//
+// INPUT:  RDI:  exception code of fault
+//         RSI:  faulting RIP
+//
+// OUTPUT:
+// 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
+NESTED_ENTRY RhpThrowHwEx, _TEXT, NoHandler
+
+        STACKSIZEOF_ExInfo = ((SIZEOF__ExInfo + 15) & (~15))
+
+        SIZEOF_OutgoingScratch  = 0x20
+        rsp_offsetof_ExInfo     = SIZEOF_OutgoingScratch
+        rsp_offsetof_Context    = SIZEOF_OutgoingScratch + STACKSIZEOF_ExInfo
+
+        mov     rax, rsp        // save the faulting RSP
+
+        add     rsi, 1  // 'faulting IP' += 1, we do this because everywhere else we treat the faulting IP as
+                        // a return-address and optionally subtract one when doing EH-related things (but not
+                        // subtracting 1 when doing GC-related things).  The fault IP here will be the start
+                        // of the faulting instruction, so +1 will point to either the next instruction or the
+                        // middle of this instruction.  Either way, when the dispatch / stackwalk code deals
+                        // with this address it'll apply a -1 for EH range checks and the GC-related operations
+                        // don't need to be precise here because the fault location isn't a GC safe point 
+
+        push_nonvol_reg r15
+        push_nonvol_reg r14
+        push_nonvol_reg r13
+        push_nonvol_reg r12
+        push_nonvol_reg rbx
+        push_register   rdx
```

Comment:
Oh, I've just realized these are on stack images of the PAL_LIMITED_CONTEXT. So I need to add one more register to those and also a padding due to that.
The pushed RDX is there is the RAX member of the context and we zero it here (if I wasn't missing the xor). I will add better comment to that.